SCClassLibrary/Common/Collections/Collection.sc

```diff
@@ -558,25 +558,31 @@ Collection {
 	asSet { ^Set.new(this.size).addAll(this) }
 	asSortedList { | function | ^SortedList.new(this.size, function).addAll(this) }
 
-	asAssociations {
+	asAssociations { |class|
 		var res;
-		if(this.isAssociationArray) { ^this };
-		res = Array.new(this.size div: 2);
+		class = class ? Array;
+		if(this.isAssociationArray) {
+			^if(class == this.class) { this } { this.as(class) }
+		};
+		res = class.new(this.size div: 2);
 		this.pairsDo { |key, val| res.add(key -> val) }
 		^res
 	}
 
-	asPairs {
+	asPairs { |class|
```

Comment:
IMO this method needs documentation, now that it has an additional parameter that can be explained.